cores/esp8266/Esp-frag.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ Esp.cpp - ESP8266-specific APIs
+ Copyright (c) 2015 Ivan Grokhotkov. All rights reserved.
+ This file is part of the esp8266 core for Arduino environment.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include "umm_malloc/umm_malloc.h"
+#include "umm_malloc/umm_malloc_cfg.h"
+#include "coredecls.h"
+#include "Esp.h"
+
+uint8_t EspClass::getHeapFragmentation (uint32_t* freeHeap, uint16_t* maxBlock)
```

Comment:
it bothers me that we have a getFreeHeap() that takes no args and return only the free heap, a getMaxBlockSize() that takes no args and returns only the max block size, and this that does take 2 optional args for free heap and maxblock and returns the fragmentation. It just seems inconsistent to me. I think I'd prefer to have 3 standalone functions one for each, plus a 4th function getHeapStats() or whatever that takes 3 args for getting everything at once (performance) and returns nothing (void).  
Thoughts?

Reply:
For the three functions (Free, MaxBlock, Fragmentation), the same `umm_info()` function is called which goes through every heap chunks. I figured that those who would call Fragmentation may optionally give pointers to fill those results for free (as this one may not be quite often used).
OK with `getHeapStats()`.